### PR TITLE
[docs] fix xref to `PreTrainedModel.generate`

### LINKED ISSUE #9202: [wanted] explicit docs for inherited methods
# 🚀 Feature request

The HF approach is to unroll most of the code for the ease of understanding, but somehow this is not the case with docs.

If possible, could we explicitly add documentation for inherited methods in the specific model documentation page? 

e.g. why https://huggingface.co/transformers/model_doc/t5.html doesn't have `T5ForConditionalGeneration.generate` documented - sure one eventually figures out it's https://huggingface.co/transformers/main_classes/model.html#transformers.generation_utils.GenerationMixin.generate but why make user's life so difficult when the docs are autogenerated anyway.

In the worst case there could be an entry for `T5ForConditionalGeneration.generate` with the link to https://huggingface.co/transformers/main_classes/model.html#transformers.generation_utils.GenerationMixin.generate, but that is not ideal since one has to leave the main doc of the model, which makes it much harder to jump around its different parts.

IMHO, it'd greatly improve user's experience to have all the functionality documentation supported by a model in one page of that model.

And xrefs are super-useful too, e.g. [t5 pre-amble](https://huggingface.co/transformers/model_doc/t5.html#overview) mentions  `T5ForConditionalGeneration.generate `but it's not linked to anywhere.

Thank you!
 
@sgugger, @LysandreJik

### PR DESCRIPTION
**Pull request by stas00**
This PR partially resolves the issue raised in https://github.com/huggingface/transformers/issues/9202

I spent quite some time to try to figure out how to get sphinx to figure out the inheritance so that it could cross-reference inherited methods, but it can't even handle mixins it seems. i.e. it can't resolve: `transformers.PreTrainedModel.generate` 

So one has to explicitly specify the fully qualified original method for xref to work :(

Bottom line - if currently you want a cross-reference link to `transformers.PreTrainedModel.generate` or `T5ForConditionalGeneration.generate` to work, you have to use `~transformers.generation_utils.GenerationMixin.generate`

So I did:
```
find . -type d -name ".git" -prune -o -type f -exec perl -pi -e 's|transformers\.\w+\.generate|transformers.generation_utils.GenerationMixin.generate|g' {} \;
```

then also added the same for `transformers.PreTrainedModel.(greedy_search|sample|beam_search|beam_sample|group_beam_search)`
```
find . -type d -name ".git" -prune -o -type f -exec perl -pi -e 's#transformers.PreTrainedModel.(greedy_search|sample|beam_search|beam_sample|group_beam_search)#transformers.generation_utils.GenerationMixin.$1#g' {} \;
```

Fixes: https://github.com/huggingface/transformers/issues/9202

@sgugger

### CODE REVIEW

File: docs/source/internal/generation_utils.rst [hunk 1]
```diff
@@ -13,7 +13,7 @@
 Utilities for Generation
 -----------------------------------------------------------------------------------------------------------------------
 
-This page lists all the utility functions used by :meth:`~transformers.PreTrainedModel.generate`,
+This page lists all the utility functions used by :meth:`~transformers.generation_utils.GenerationMixin.generate`,
 :meth:`~transformers.PreTrainedModel.greedy_search`, :meth:`~transformers.PreTrainedModel.sample`,
 :meth:`~transformers.PreTrainedModel.beam_search`, :meth:`~transformers.PreTrainedModel.beam_sample`, and
 :meth:`~transformers.PreTrainedModel.group_beam_search`.
```
Comment (sgugger) [inline docs/source/internal/generation_utils.rst:19]: I think those should point to the GenerationMixin as well, as it's the place where they come from and they are probably not resolved either.

**Review by sgugger — APPROVED** [review]
Thanks for fixing this! I think ti can be applied to the other methods like generate (though not sure if they are referenced much in the lib).

**Comment by stas00** [comment]
Would it be a good idea to make an Good second Issue for someone to research whether sphinx could be configured to resolve inherited and mixin methods when it does cross-referencing? It'd be much more intuitive for doc writing if we could have the sub-classed method named and also not needing to hide the sub-class via `~` prefix.

I did spent quite some time trying to find one and also experimenting with different configs but it doesn't mean it doesn't exist.

**Comment by sgugger** [comment]
Sure, we can try that. I didn't try for very long but didn't manage it on my side either, for what it's worth.

**Comment by stas00** [comment]
https://github.com/huggingface/transformers/issues/11069

**Comment by stas00** [comment]
https://github.com/huggingface/transformers/issues/11069 had many fishermen but nothing came out of it. So I guess I will merge it and close https://github.com/huggingface/transformers/issues/11069

**Comment by sgugger** [comment]
Fine by me!

**Pull request merged**